### PR TITLE
Move API OpenID-Connect support to Apache configuration

### PR DESCRIPTION
**Pull request by jvlcek**
Fixes ManageIQ/manageiq#19866

This PR, combined with others in other github repos, will move the support for the ManageIQ API out of code and into the Apache configuration.

**Dependent PRs**
https://github.com/ManageIQ/manageiq-appliance/pull/282
https://github.com/ManageIQ/manageiq/pull/20131
https://github.com/ManageIQ/manageiq-appliance_console/pull/117

### CODE REVIEW

**Comment by jvlcek** [comment]
@miq-bot add_label wip
Setting wip label until more testing is done.

File: app/controllers/api/base_controller/authentication.rb [hunk 1]
```diff
@@ -147,7 +147,7 @@ def valid_ui_session?
 
       def authenticate_with_jwt
         timeout = ::Settings.api.authentication_timeout.to_i_with_method
```
Comment (chessbyte) [inline app/controllers/api/base_controller/authentication.rb:149]: the setting of the timeout now is not needed, right?

**Comment by abellotti** [comment]
@jvlcek do you have an example Authentication and Authorization lines from the api.log for a JWT auth ?   Thanks.

File: app/controllers/api/base_controller/authentication.rb (hunk 1 above)
Comment (jvlcek) [reply app/controllers/api/base_controller/authentication.rb:149]: Good catch @chessbyte,  but backwards. I had avoided using the timeout to aid in getting things working. The timeout should be used, adding use of the timeout back.

**Comment by jvlcek** [comment]
> @jvlcek do you have an example Authentication and Authorization lines from the api.log for a JWT auth ? Thanks.

```log
[----] I, [2020-05-06T16:09:18.630446 #15779:f03e44]  INFO -- : MIQ(Api::UsersController.log_request_initiated)  
[----] I, [2020-05-06T16:09:18.630644 #15779:f03e44]  INFO -- : MIQ(Api::UsersController.log_request) API Request:    {:requested_at=>"2020-05-06 20:09:18 UTC", :method=>"GET", :url=>"https://joev-oidc.jvlcek.redhat.com/api/users"}
[----] I, [2020-05-06T16:09:18.673310 #15779:f03e44]  INFO -- : MIQ(Api::UsersController.log_request) Authentication: {:type=>"jwt", :token=>nil, :x_miq_group=>nil, :user=>"jvlcek@jvlcek.redhat.com"}
[----] I, [2020-05-06T16:09:18.675832 #15779:f03e44]  INFO -- : MIQ(Api::UsersController.log_request) Authorization:  {:user=>"jvlcek@jvlcek.redhat.com", :group=>"EvmGroup-super_administrator", :role=>"EvmRole-super_administrator", :tenant=>"My Company"}
[----] I, [2020-05-06T16:09:18.676293 #15779:f03e44]  INFO -- : MIQ(Api::UsersController.log_request) Request:        {:method=>:get, :action=>"read", :fullpath=>"/api/users", :url=>"https://joev-oidc.jvlcek.redhat.com/api/users", :base=>"https://joev-oidc.jvlcek.redhat.com", :path=>"/api/users", :prefix=>"/api", :version=>"4.3.0-pre", :api_prefix=>"https://joev-oidc.jvlcek.redhat.com/api", :collection=>"users", :c_suffix=>nil, :collection_id=>nil, :subcollection=>nil, :subcollection_id=>nil}
[----] I, [2020-05-06T16:09:18.676650 #15779:f03e44]  INFO -- : MIQ(Api::UsersController.log_request) Parameters:     {"action"=>"index", "controller"=>"api/users", "format"=>"json", "body"=>{}}
[----] I, [2020-05-06T16:09:18.681873 #15779:f03e44]  INFO -- : MIQ(Api::UsersController.log_request) Response:       {:completed_at=>"2020-05-06 20:09:18 UTC", :size=>"0.939 KBytes", :time_taken=>"0.051 Seconds", :status=>200}
```

**Comment by jvlcek** [comment]
@miq-bot remove_label wip

**Comment by simaishi** [comment]
Jansa backport details:

```
$ git log -1
commit ff4769717e48f5113fbd089558ce68f5bf59c81b
Author: Jason Frey <fryguy9@gmail.com>
Date:   Fri May 8 16:34:19 2020 -0400

    Merge pull request #828 from jvlcek/oidc_to_httpd_config_issue_19866

    Move API OpenID-Connect support to Apache configuration

    (cherry picked from commit da56aabc6e7728d341150362085c641adf24ba71)
```

**Pull request merged**